protos/foundation/protos/doc/entity.proto

```diff
@@ -109,54 +121,61 @@ message Phrase {
 */
 message Number {
 	required Token token = 1;
-	optional double value = 2;
+	required geometry.BBox bbox = 2;
+	optional double value = 3;
 }
 
 message Integer {
 	required Token token = 1;
-	optional int64 value = 2;
+	required geometry.BBox bbox = 2;
+	optional int64 value = 3;
 }
 
 message Date {
 	required Token token = 1;
+	required geometry.BBox bbox = 2;
 	// Let this value be an ISO8601 formatted date.
 	// TODO: we should settle on an appropriate denotation format here.
-	optional string value = 2;
+	optional string value = 3;
 }
 
 message Time {
 	required Token token = 1;
+	required geometry.BBox bbox = 2;
 	// Let this value be UNIX time in seconds from the epoch.
 	// TODO: we should settle on an appropriate denotation format here.
-	optional uint64 value = 2;
+	optional uint64 value = 3;
 }
 
 // TODO: do we want a Datetime type?
 
 message Currency {
 	required Token token = 1;
+	required geometry.BBox bbox = 2;
 	// Fixed-point representation for currency avoids float-rounding issues.
 	// TODO: we should settle on an appropriate denotation format here.
```

Comment:
This is easier if we don't bundle all currency types into a single type. That would be my vote.

What are the advantages of having a single currency type?

Reply:
Most operations would probably be similar, as long as units are equivalent. Like, arithmetic would stay the same, comparisons/equality, etc. But if we broke out Currency into different types, with their denotation type being the same, that would probably be better than having to check the `units` field for every operation.